tools/bridge/bridge/confirmation_task_planner.py

```diff
@@ -74,7 +82,72 @@ def pull_transfers_to_confirm(self, current_time: float) -> List[AttributeDict]:
                 - self.scheduled_hashes
             )
             self.scheduled_hashes |= unconfirmed_transfer_hashes
-            return [
+            confirmation_tasks = [
                 self.transfer_events[transfer_hash]
                 for transfer_hash in unconfirmed_transfer_hashes
             ]
+
+        self.clear_transfers()
+        return confirmation_tasks
+
+
+class ConfirmationTaskPlanner:
+    def __init__(
+        self,
+        sync_persistence_time: float,
+        transfer_event_queue: Queue,
+        home_bridge_event_queue: Queue,
+        confirmation_task_queue: Queue,
+    ) -> None:
+        self.logger = logging.getLogger(
+            "bridge.confirmation_task_planner.ConfirmationTaskPlanner"
+        )
+
+        self.recorder = TransferRecorder(sync_persistence_time)
+
+        self.transfer_event_queue = transfer_event_queue
+        self.home_bridge_event_queue = home_bridge_event_queue
+
+        self.confirmation_task_queue = confirmation_task_queue
+
+    def run(self):
+        self.logger.info("Starting")
+        try:
+            greenlets = [
+                gevent.spawn(self.process_transfer_events),
+                gevent.spawn(self.process_home_bridge_events),
+            ]
+            gevent.joinall(greenlets, raise_error=True)
+        finally:
+            self.logger.info("Stopping")
+            for greenlet in greenlets:
+                greenlet.kill()
+
+    def process_transfer_events(self) -> None:
+        while True:
+            event = self.transfer_event_queue.get()
+            if isinstance(event, FetcherReachedHeadEvent):
+                self.logger.info("Transfer events are in sync now")
+            else:
+                self.logger.info("Received transfer to confirm")
+                self.recorder.apply_proper_event(event)
+                self.check_for_confirmation_tasks()
+
+    def process_home_bridge_events(self) -> None:
+        while True:
+            event = self.home_bridge_event_queue.get()
+            if isinstance(event, FetcherReachedHeadEvent):
+                self.logger.info("Home bridge is in sync now")
+                self.recorder.apply_home_chain_synced_event(event.timestamp)
+            else:
+                self.logger.info("Received home bridge event")
+                self.recorder.apply_proper_event(event)
+            self.check_for_confirmation_tasks()
```

Comment:
Why don't we move this to the end of the first if block? To me it's rather surprising why it's being called for every processed event.

Reply:
Looking over the code I see it's also being called from `process_transfer_events`. 

I would like to keep this as simple as possible.

Please only call this when you see the `FetcherReachedHeadEvent` on the home chain.

Reply:
i.e. do not call it from `process_transfer_events` and move this into the first if block.